### PR TITLE
feat: add CORS instance settings

### PR DESCRIPTION
**Pull request by olav**
https://linear.app/unleash/issue/1-171/embedded-proxy-instance-cors-settings-api

https://github.com/Unleash/unleash/issues/1875

### CODE REVIEW

File: src/lib/util/validateOrigin.test.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    expect(validateOrigin('**')).toEqual(false);
+    expect(validateOrigin('localhost')).toEqual(false);
+    expect(validateOrigin('localhost:8080')).toEqual(false);
+    expect(validateOrigin('//localhost:8080')).toEqual(false);
+    expect(validateOrigin('http://localhost/')).toEqual(false);
+    expect(validateOrigin('http://localhost/a')).toEqual(false);
+    expect(validateOrigin('https://example.com/a')).toEqual(false);
+    expect(validateOrigin('https://example.com ')).toEqual(false);
+
+    expect(validateOrigin('*')).toEqual(true);
```
Comment (Tymek) [inline src/lib/util/validateOrigin.test.ts:16]: is `*.localhost` valid? or `https://*.example.com`?

Comment (olav) [reply src/lib/util/validateOrigin.test.ts:16]: Nope, for CORS we have to use `*` or a full origin. Would be nice though!

Comment (Tymek) [reply src/lib/util/validateOrigin.test.ts:16]: Could be added to as expected to fail

**Review by Tymek — APPROVED** [review]

File: src/lib/util/validateOrigin.test.ts (hunk 1 above)
Comment (olav) [reply src/lib/util/validateOrigin.test.ts:18]: Yeah, good idea!

**Pull request merged**